include/random_walks/sparse_uniform_billiard_walk.hpp

```diff
@@ -0,0 +1,256 @@
+        if (Chol.info() != Eigen::Success) {
+            throw std::runtime_error("Sparse Cholesky decomposition failed");
+        }
+        
+        _L_inv = Chol.matrixL().transpose();
+        
+        MT A_dense = _A.toDense();
+        MT A_transposed = A_dense.transpose();
+        MT temp = _L_inv.template triangularView<Eigen::Upper>().solve(A_transposed);
+        _A_rounded = temp.transpose();
+        
+        _A_rounded_row_norms.setZero(_A_rounded.rows());
+        NT* A_rounded_row_norms_data = _A_rounded_row_norms.data();
+        for (int i = 0; i < _A_rounded.rows(); ++i) {
+            NT row_norm = _A_rounded.row(i).norm();
+            *A_rounded_row_norms_data = row_norm;
+            _A_rounded.row(i) /= row_norm;
+            A_rounded_row_norms_data++;
+        }
+    }
+
+    template <typename GenericPolytope>
+    void initialize(GenericPolytope& P,
+                    const Point& p_rounded,
+                    RandomNumberGenerator& rng)
+    {
+        unsigned int n = P.dimension();
+        const NT dl = 0.995;
+        
+        _p = p_rounded;
+        _v = GetDirection<Point>::apply(n, rng);
+                
+        _Ar.setZero(_A.rows());
+        _Av.setZero(_A.rows());
+        _lambda_prev = 0;
+        
+        NT T = rng.sample_urdist() * _Len;
+        
+        auto pbpair = P.sparse_line_positive_intersect(_p, _v, *_oracle_params);
+        
+        if (pbpair.second < 0) {
+            _p += T * _v;
+            _lambda_prev = T;
+            return;
+        }
+        
+        if (T <= pbpair.first) {
+            _p += (T * _v);
+            _lambda_prev = T;
+            return;
+        }
+        
+        _lambda_prev = dl * pbpair.first;
+        _p += (_lambda_prev * _v);
+        T -= _lambda_prev;
+        
+        P.sparse_compute_reflection(_v, *_oracle_params);
+        VT p_round = _p.getCoefficients();
+        _Ar = _A * _L_inv.transpose().template triangularView<Eigen::Lower>().solve(p_round);
+        _lambda_prev = 0;
+        
+        int it = 0;
+        while (it <= 50*n && T > 0)
+        {
+            auto pbpair2 = P.sparse_line_positive_intersect(_p, _v, _Ar, _Av, _lambda_prev, *_oracle_params);
+            
+            if (T <= pbpair2.first) {
+                _p += (T * _v);
+                _lambda_prev = T;
+                break;
+            } else if (it == 50*n) {
+                _lambda_prev = rng.sample_urdist() * pbpair2.first;
+                _p += (_lambda_prev * _v);
+                break;
+            }
+            
+            _lambda_prev = dl * pbpair2.first;
+            _p += (_lambda_prev * _v);
+            T -= _lambda_prev;
+            
+            P.sparse_compute_reflection(_v, *_oracle_params);
+            VT p_round = _p.getCoefficients();
+            _Ar = _A * _L_inv.transpose().template triangularView<Eigen::Lower>().solve(p_round);
+            _lambda_prev = 0;
```

Comment:
why do you set it to 0? Plz delete this. The Ar should be updated in the next oracle call using this intersection time. Plz go through the other billiard walk structs carefully and try to mimic their update logic